src/client.ts

```diff
@@ -1063,4 +1066,35 @@ export class SpvWalletClient {
     }
     return validateTotpForContact(this.xPrivKey, contact, passcode, requesterPaymail, period, digits);
   }
+
+  /**
+   * Syncs merkleroots from the client db to the last known block by SPV-Wallet
+   *
+   * @param {Repository} repo - Repository interface capable of reading lastEvaluatedKey and saving to the database
+   * @returns void
+   */
+  async SyncMerkleRoots(repo: Repository) {
+    let lastEvaluatedKey = await repo.getLastEvaluatedKey();
+    let requestPath = 'merkleroots';
+    let lastEvaluatedKeyQuery = '';
+
+    if (lastEvaluatedKey) {
+      lastEvaluatedKeyQuery = `?lastEvaluatedKey=${lastEvaluatedKey}`;
+    }
+
+    while (true) {
+      const merkleRootsResponse: ExclusiveStartKeyPage<MerkleRoot[]> = await this.http.request(
+        `${requestPath}${lastEvaluatedKeyQuery}`,
+        'GET',
+      );
+
+      await repo.saveMerkleRoots(merkleRootsResponse.content);
+
+      if (merkleRootsResponse.page.lastEvaluatedKey === '') {
+        break;
+      }
+
+      lastEvaluatedKeyQuery = `?lastEvaluatedKey=${merkleRootsResponse.page.lastEvaluatedKey}`;
+    }
```

Comment:
Yeah I like the idea of no. 3. I was thinking about a timeout or number of iterations, but the idea of this endpoint is also to allow a client that doesn't have any merkleroots at all to sync with everything from spv-wallet. Currently the tip on WOC is #864761 thats approximately 433 iterations. 

Now the thing is what would be that large number to prevent iterations or what would be a good timeout, since in the case I mentioned above it can take a couple of minutes for sure. If you've got an idea I'm all ears and I'd implement it here

Reply:
About the timeout.
I suppose - in the client's application this function will/might be called in some cron-job task. This timeout could be set to slightly less than the cron-job period to prevent overlaping. 
Think about it 😄 🙈